File: ACT8/act8.pl
/*Definición de la regla para calcular la potencia
    pow(X,N, Res).  
    pow(X,5, Res).
    pow(6,N, R).
    */
/* Caso base: Cualquier número elevado a la potencia 0 es 1. */
pow(_, 0, 1).

/* Caso recursivo: Elevar X a la potencia N */
pow(X, N, Res) :-
    N > 0,                  
    N1 is N - 1,            
    pow(X, N1, Temp),       
    Res is X * Temp.  

/* Aseguramos que N sea un número positivo */
/* Reducimos N en 1 en cada llamada recursiva */
/* Llamada recursiva */
/* Calculamos el Resultado */

File: ACT8/act8.py
def pot(base, exp):
    # Caso base: Cualquier número elevado a la potencia 0 es 1.
    if exp == 0:
        return 1
    # Caso recursivo: Si el exponente es positivo.
    elif exp > 0:
        return base * pot(base, exp - 1)
    # Caso recursivo: Si el exponente es negativo.
    else:
        return 1 / (base * pot(base, -exp - 1))

# Solicitar al usuario la base y el exponente
base = float(input("Ingrese la base: "))
exp = int(input("Ingrese el exponente: "))

# Calcular la pot y mostrar el resultado
resultado = pot(base, exp)
print(f"{base} elevado a la {exp} es igual a {resultado}")
